src/Microsoft.AspNet.Mvc.ModelBinding/Binders/MutableObjectModelBinder.cs

```diff
@@ -90,16 +95,7 @@ protected virtual bool CanUpdateProperty(ModelMetadata propertyMetadata)
                 return true;
             }
 
-            // 3. The model name is not prefixed and a value provider can directly provide a value for the model name.
```

Comment:
> this relate to the fix that we are looking at

it _is_ the fix

> Complex objects can be directly bound by the value providers

what value providers do this?  even if such exist or someone could write one, `MutableObjectModelBinder` can't handle the case -- it always creates `ComplexModelDto` and binds the individual properties.

> Can you point me to those scenarios?

cases involving `BinderModelName` mostly. look at the special cases in `CanValueBindAnyModelProperties` and `CanBindValue`